src/pages/example--hello-world.mdx

```diff
@@ -185,82 +178,64 @@ import * as cbor from "https://deno.land/x/cbor@v1.4.1/index.js";
 const lucid = await Lucid.new(
   new Blockfrost(
     "https://cardano-preview.blockfrost.io/api/v0",
-    /* BLOCKFROST API KEY */,
+    Deno.env.get("BLOCKFROST_API_KEY")
   ),
-  "Preview",
+  "Preview"
 );
+```
 
+### Reading The Blueprint
+
+Next, we'll need to read the validator from the blueprint we generated earlier. We'll also need to convert it to a format that Lucid understands. This is done by encoding the validator as CBOR and then converting it to hex as shown below:
+
+```ts filename="hello_world-lock.ts"
 lucid.selectWalletFromPrivateKey(await Deno.readTextFile("./key.sk"));
 
 const validator = await readValidator();
 
 // --- Supporting functions
 
 async function readValidator(): Promise<SpendingValidator> {
-  const validator = JSON.parse(await Deno.readTextFile("plutus.json")).validators[0];
+  const validator = JSON.parse(await Deno.readTextFile("plutus.json"))
+    .validators[0];
   return {
     type: "PlutusV2",
     script: toHex(cbor.encode(fromHex(validator.compiledCode))),
   };
 }
 ```
 
-<Callout>
-  If you've installed [deno](https://deno.land/manual@v1.29.1/getting_started/installation), you can run the excerpt above by executing:
+### Finishing The Locking Contract Logic
 
-```
-deno run --allow-net --allow-read hello_world-lock.ts
-```
-
-It assumes that this file (`hello_world-lock.ts`) is placed at the root of your `hello_world` folder. At this stage, your folder should looks roughly like this:
-
-```
-./hello_world
-│
-├── README.md
-├── aiken.toml
-├── plutus.json
-├── hello_world-lock.ts
-├── key.addr
-├── key.sk
-├── lib
-│   └── ...
-└── validators
-    └── hello_world.ak
-```
-
-</Callout>
-
-### Locking funds into the contract
-
-Here, we make our first transaction to lock funds into the contract. The datum
+Now that we can read our validator, we can make our first transaction to lock funds into the contract. The datum
 must match the representation expected by the script, so this is a constructor
 with a single field that is a byte array.
 
 As value for that byte array, we provide a hash digest of our public key. This
 will be needed to unlock the funds.
 
 ```ts filename="hello_world-lock.ts"
-// ^^^ Code above is unchanged. ^^^
-
 const publicKeyHash = lucid.utils.getAddressDetails(
   await lucid.wallet.address()
-).paymentCredential.hash;
+).paymentCredential?.hash;
 
 const datum = Data.to(new Constr(0, [publicKeyHash]));
 
-const txLock = await lock(1000000, { into: validator, owner: datum });
+const txHash = await lock(BigInt(1000000), { into: validator, owner: datum });
 
-await lucid.awaitTx(txLock);
+await lucid.awaitTx(txHash);
 
-console.log(`1 ADA locked into the contract
-    Tx ID: ${txLock}
+console.log(`1 ADA locked into the contract at:
```

Comment:
at: is a bit unnecessary here

Reply:
seems like a nit to me

Reply:
Please revert that one. "Transaction ID" / "Tx ID" is still much more common in people's head. We want to limit the amount of friction in this tutorial.